STYLE_GUIDE.md

```diff
@@ -0,0 +1,1245 @@
+
+#### Bad
+
+```
+<===>
+CSS requires at least one token in a custom property.
+<===> empty/input.scss
+.empty {
+  --property:;
+}
+
+<===> empty/error
+Error: Expected token.
+  ,
+3 |   --property:;
+  |              ^
+  '
+  /sass/spec/css/custom_properties/error/empty/input.scss 3:14  root stylesheet
+```
+
+</details>
+
+HRX supports its own comment syntax, but that should never be used for specs.
+Instead [use Sass comments][] to document an individual file or spec and
+[`README.md` files][] to document entire directories.
+
+[use Sass comments]: #consider-adding-a-comment-explaining-your-spec
+[`README.md` files]: #consider-adding-readme-md-files-to-parent-directories
+
+### CONSIDER adding README.md files to parent directories
+
+<details>
+<summary>Example</summary>
+
+#### Good
+
+```hrx
+<===> color/literal/README.md
+When colors are written literally by the user, rather than being generated by a
+function, `inspect()` should return them in the same format the user wrote them.
+
+<===>
+================================================================================
+<===> color/literal/short_hex/input.scss
+a {
+  $result: inspect(#00f);
+  result: $result;
+  type: type-of($result);
+}
+
+<===> color/literal/short_hex/output.css
+a {
+  result: #00f;
+  type: string;
+}
+
+<===>
+================================================================================
+<===> color/literal/long_hex/input.scss
+a {
+  $result: inspect(#0000ff);
+  result: $result;
+  type: type-of($result);
+}
+
+<===> color/literal/long_hex/output.css
+a {
+  result: #0000ff;
+  type: string;
+}
+```
+
+</details>
+
+Like [adding a comment][] to an individual spec, add a `README.md` file to a
+directory to explain something about all the specs that directory contains.
+There's not currently a way to render the Markdown, but it provides a consistent
+format that's easy to read in plain text.
+
+[adding a comment]: #consider-adding-a-comment-explaining-your-spec
+
+## Formatting
+### DO use comment dividers to separate specs within an HRX file
+
+<details>
+<summary>Example</summary>
+
+#### Good
+
+```hrx
+<===> single/input.scss
+a {
+  b {c: d}
+}
+
+<===> single/output.css
+a b {
+  c: d;
+}
+
+<===>
+================================================================================
+<===> double/input.scss
+a {
+  b, c {d: e}
+}
+
+<===> double/output.css
+a b, a c {
+  d: e;
+}
+```
+
+#### Bad
+
+```hrx
+<===> single/input.scss
+a {
+  b {c: d}
+}
+
+<===> single/output.css
+a b {
+  c: d;
+}
+
+<===> double/input.scss
+a {
+  b, c {d: e}
+}
+
+<===> double/output.css
+a b, a c {
+  d: e;
+}
+```
+
+</details>
+
+The first file of each spec within an HRX file other than the first should begin with:
+
+```hrx
+<===>
+================================================================================
+```
+
+This is an HRX comment, and has no effect on the semantics of the file. It just
+serves to visually separate specs from one another.
```

Comment:
Are these a fixed length? Is there an HRX spec where they are described you could link to?